pkg/apis/kubebind/v1alpha1/apiservicebinding_types.go

```diff
@@ -87,8 +87,265 @@ type APIServiceBindingSpec struct {
+	// Mutually exclusive with autoDonate.
+	//
+	// +optional
+	AutoAdopt bool `json:"autoAdopt,omitempty"`
+
+	// autoDonate set to true means that a newly created object by the provider is immediately owned by the consumer.
+	// If false, the object stays in ownership of the provider.
+	// Mutually exclusive with autoDonate.
+	//
+	// +optional
+	AutoDonate bool `json:"autoDonate,omitempty"`
+
+	// onConflict determines how the conflicts between objects on the consumer cluster will be resolved.
+	//
+	// +optional
+	// +kubebuilder:default:={}
+	OnConflict *OnConflictOptions `json:"onConflict,omitempty"`
+
+	// update lists which updates to objects on the consumer cluster are claimed.
+	// By default, the whole object is synced, but metadata is not.
+	//
+	// +optional
+	Update *UpdateOptions `json:"update,omitempty"`
+}
+
+type ReadOptions struct {
+	// labels is a list of claimed label key wildcard patterns
+	// that are synchronized from the consumer cluster to the provider on
+	// objects that are owned by the consumer.
+	//
+	// +optional
+	Labels []Matcher `json:"labels,omitempty"`
+
+	// labelsOnProviderOwnedObjects is a list of claimed label key wildcard
+	// patterns that are synchronized from the consumer cluster
+	// to the provider on objects owned by the provider.
+	//
+	// +optional
+	LabelsOnProviderOwnedObjects []Matcher `json:"labelsOnProviderOwnedObjects,omitempty"`
+
+	// annotations is a list of claimed annotation key wildcard patterns
+	// that are synchronized from the consumer cluster to the provider on
+	// objects that are owned by the consumer.
+	//
+	// +optional
+	Annotations []Matcher `json:"annotations,omitempty"`
+
+	// overrideAnnotations is a list of claimed annotation key wildcard
+	// patterns that are synchronized from the consumer cluster
+	// to the provider on objects owned by the provider.
+	//
+	// +optional
+	OverrideAnnotations []Matcher `json:"overrideAnnotations,omitempty"`
+}
+
+type Matcher struct {
+	// +optional
+	Pattern string `json:"pattern,omitempty"`
+}
+
+type OnConflictOptions struct {
+	// recreateWhenConsumerSideDeleted set to true (the default) means the provider will recreate the object
+	// in case the object is missing on the consumer cluster, but has been synchronized before.
+	//
+	// If set to false, deleted provider-owned objects get deleted on the provider cluster as well.
+	//
+	// Even if the consumer mistakenly or intentionally
+	// deletes the object, the provider will recreate it. If the field is set as false,
+	// the provider will not recreate the object in case the object is deleted on the consumer cluster.
+	//
+	// +kubebuilder:default:=true
+	RecreateWhenConsumerSideDeleted bool `json:"recreateWhenConsumerSideDeleted"`
+}
+
+type CreateOptions struct {
+	// replaceExisting means that an existing object owned by the consumer will be replaced by the provider object.
+	//
+	// If set to false, and a conflicting consumer object exists, it is not touched.
```

Comment:
nit:
```suggestion
	// If not true, and a conflicting consumer object exists, it is not touched.
```
Formally, false and unset are different values. Actually, we could also default this to `false` and remove the omitempty.